docs/aws/audit/redshiftmonitoring/rules/redshift_nodes_counts_remediation.mdx

```diff
@@ -1,6 +1,269 @@
+       )
+       return response
+
+   # Example usage
+   cluster_identifier = 'YOUR_CLUSTER_IDENTIFIER'
+   new_node_count = 2  # Adjust based on your scaling policy
+   response = scale_redshift_cluster(cluster_identifier, new_node_count)
+   print(response)
+   ```
+
+By following these steps, you can proactively monitor and manage your Redshift nodes to prevent hitting the node limit.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console and open the Amazon Redshift console at https://console.aws.amazon.com/redshift/.
+   
+2. In the navigation pane, choose "Clusters". This will open a list of all your existing Redshift clusters.
+
+3. Click on the name of the cluster you want to check. This will open the cluster details page.
+
+4. In the "Configuration" tab, look for the "Nodes" section. Here, you can see the number of nodes currently in use for the selected cluster. Compare this number with the maximum limit for your account to determine if you are close to or exceeding the limit.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all Redshift clusters: Use the following command to list all the Redshift clusters in your AWS account.
+
+   ```
+   aws redshift describe-clusters
+   ```
+
+   This command will return a JSON output with details of all the Redshift clusters.
+
+3. Check the number of nodes in each cluster: In the JSON output from the previous step, look for the 'NumberOfNodes' field. This field indicates the number of nodes in the cluster. 
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier,NumberOfNodes]'
+   ```
+
+   This command will return a list of all clusters along with their node count.
+
+4. Compare the number of nodes with the limit: AWS has a default limit of 50 nodes per cluster for Redshift. If the 'NumberOfNodes' field in the output is close to or at this limit, then you are at the Redshift nodes limit.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3): To interact with AWS services, you need to install and configure Boto3. You can install it using pip:
+
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials to enable Boto3 to communicate with AWS services. You can do this by setting the following environment variables:
+
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>